Integrating das branch into master

lucassaldanha opened the pull request:
## PR Description
Integrating das branch into master.

## Fixed Issue(s)
N/A
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

lucassaldanha committed:
DAS magic

lucassaldanha committed:
Merge branch 'das-squashed' into master-das

lucassaldanha committed:
fixes

lucassaldanha commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java:
```diff
@@ -552,18 +569,19 @@ private BlockImportResult importBlockAndState(
       case INVALID -> {
-        debugDataDumper.saveInvalidBlobSidecars(
-            blobSidecarsAndValidationResult.getBlobSidecars(), block);
+        LOG.error("sidecars validation result: {}", dataAndValidationResult::toLogString);
```
@zilm13 do we want to keep this as ERROR?

zilm13 replied:
debug I think

lucassaldanha committed:
cleaning logs and unifying TODOs

lucassaldanha committed:
more log cleanning :)

lucassaldanha commented on networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/GossipTopicDasPeerCustodyTracker.java:
```diff
@@ -0,0 +1,120 @@
+import tech.pegasys.teku.spec.datastructures.state.ForkInfo;
+import tech.pegasys.teku.statetransition.datacolumns.retriever.DasPeerCustodyCountSupplier;
+
+public class GossipTopicDasPeerCustodyTracker
```
@zilm13 it looks like this isn't used anywhere... have we missed something?

zilm13 replied:
ahh yeah, it's hacky way, we get all peer subscriptions and determinate subnets by its subscription
we must drop it since info was added to enr and metadata

lucassaldanha committed:
Added ticket urls to TODO-fulu

rolfyone commented on beacon/validator/src/test/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactoryTest.java:
```diff
@@ -683,284 +637,6 @@ void shouldUnblindSignedBlindedBeaconBlock(final boolean useLocalFallback) {
-        .hasSameElementsAs(blobsBundle.getCommitments());
-  }
-
-  @Test
-  void shouldIncludeKzgCommitmentsInBlindedBlock() {
-    final BeaconState blockSlotState = dataStructureUtil.randomBeaconState();
```
this kinda looks accidental

lucassaldanha replied:
These tests have been moved to `BlockOperationSelectorFactoryTestDeneb` and `BlockOperationSelectorFactoryTestFulu` respectively.

zilm13 replied:
this test was split to 4 separate tests, it was probably moved to another

zilm13 replied:
I see it in das branch in `BlockOperationSelectorFactoryTestDeneb` and `BlockOperationSelectorFactoryTestFulu`

zilm13 committed:
Fix Fulu block production (#9481)

lucassaldanha committed:
Merge branch 'master' into master-das

jeroost committed:
Post Fulu BlobSidecars RPC deprecation (#9408)

lucassaldanha committed:
Merge branch 'master' into master-das

lucassaldanha committed:
Only creating DasPreSample when FULU is scheduled

lucassaldanha committed:
PR comments

rolfyone committed:
cleanup NPE on start

also fixed `initDasCustody` a bit
 - missing finals
 - random anonymous code segments

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
oopsie

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
cleanup

start on some of the codeql comments

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
more cleanup

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
more cleanup

Signed-off-by: Paul Harris <paul.harris@consensys.net>

lucassaldanha committed:
Revert changes in EventLogger

lucassaldanha committed:
Merge branch 'master-das' of github.com:lucassaldanha/teku into master-das

rolfyone committed:
should be debug

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'lucassaldanha/master-das' into master-das

zilm13 commented on acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/SimpleHttpClient.java:
```diff
@@ -58,19 +69,32 @@ public Bytes getAsBytes(final URI baseUrl, final String path, final Map<String,
     headers.forEach(builder::header);
     final Response response = httpClient.newCall(builder.build()).execute();
     final ResponseBody body = response.body();
     if (!response.isSuccessful()) {
```
nit: It's more readable when not inverted

zilm13 commented on beacon/sync/src/test/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/PeerSyncTest.java:
```diff
@@ -551,6 +554,68 @@ void sync_blocksAndBlobSidecarsForDeneb() {
+
+    // Check that the sync is done and the peer was not disconnected.
+    assertThat(syncFuture).isCompleted();
+    verify(peer, never()).disconnectCleanly(any());
```
we could add `disconnectImmediately` check if paranoid also

zilm13 commented on beacon/sync/src/test/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/PeerSyncTest.java:
```diff
@@ -551,6 +554,68 @@ void sync_blocksAndBlobSidecarsForDeneb() {
+
+    completeRequestWithBlocksAtSlots(blocksRequestFuture, denebSecondSlot, denebPeerSlotsAhead);
+
+    verifyBlobSidecarsAddedToPool(denebSecondSlot, slotsPerEpochMinimal, blobSidecarsBySlot);
```
I don't like this test. It delivers last BlobSidecar in a Fulu epoch, which shouldn't happen. Probably it's set with `when` but it shouldn't be like this anyway. I'd address it in a separate issue

zilm13 commented on beacon/sync/src/test/java/tech/pegasys/teku/beacon/sync/forward/singlepeer/PeerSyncTest.java:
```diff
@@ -551,6 +554,68 @@ void sync_blocksAndBlobSidecarsForDeneb() {
+    when(recentChainData.getFinalizedEpoch()).thenReturn(denebPeerFinalizedEpoch);
+
+    final UInt64 slotsPerEpochMinimal = UInt64.valueOf(8);
+    final UInt64 denebLastSlot = denebFirstSlot.plus(slotsPerEpochMinimal);
```
it's in FULU, not in deneb, there is even a check below

zilm13 commented on beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java:
```diff
@@ -257,7 +257,7 @@ private boolean isInHistoricBlockRoots(
         "Checking if block {} is in historic block roots {} of the state {}",
         producedBlock,
-        state.getBlockRoots(),
+        state.getBlockRoots().hashTreeRoot(),
```
I think we could remove this parameter completely, not sure it makes any sense

zilm13 commented on beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java:
```diff
@@ -163,7 +236,14 @@ abstract SafeFuture<Void> publishBlock(
   abstract void publishBlobSidecars(
-      List<BlobSidecar> blobSidecars, BlockPublishingPerformance blockPublishingPerformance);
+      List<BlobSidecar> blobSidecars,
+      SignedBeaconBlock block,
```
we don't use block anywhere

zilm13 commented on beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java:
```diff
@@ -163,7 +236,14 @@
+
+  abstract void publishDataColumnSidecars(
+      List<DataColumnSidecar> dataColumnSidecars,
+      SignedBeaconBlock block,
```
Computer is  right, we don't use block and don't plan to

zilm13 commented on beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java:
```diff
@@ -323,24 +342,46 @@ private SafeFuture<Void> setKzgCommitments(
-              .orElseThrow()
-              .thenApply(Optional::orElseThrow)
-              .thenApply(blobKzgCommitmentsSchema::createFromBlobsBundle);
+    if (bodyBuilder.supportsCellProofs()) {
```
Hell. I think the whole class could be easily refactored when every setter case and selector is a separate class. Maybe create an issue for this?

zilm13 commented on data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/addon/FuluRestApiBuilderAddon.java:
```diff
@@ -0,0 +1,46 @@
+
+  @Override
+  public boolean isEnabled() {
+    return spec.isMilestoneSupported(SpecMilestone.ELECTRA);
```
FULU?

zilm13 commented on eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/fulu/networking/GetCustodyGroupsTestExecutor.java:
```diff
@@ -0,0 +1,68 @@
+    private String nodeId;
+
+    @JsonProperty(value = "custody_group_count", required = true)
+    private int custodySubnetCount;
```
CustodyGroupCount and +2

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/BeaconBlockInvariants.java:
```diff
@@ -52,8 +52,8 @@ public static UInt64 extractBeaconBlockSlot(final Bytes bytes) {
    * offset to the start of the beacon block data is recorded. Use that prefix to get the {@link
-   * BeaconBlock} data or in case of {@link SignedBlockContents} the {@link SignedBeaconBlock} data
-   * and then find the slot as for an unsigned block.
+   * BeaconBlock} data or in case of {@link SignedBlockContentsDeneb} the {@link SignedBeaconBlock}
```
it's not only for SignedBlockContentsDeneb, it will work for fulu, worth rephrasing and dropping link, maybe just call it  `SignedBlockContents`without a link as it's a spec object

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/BlockContainer.java:
```diff
@@ -20,12 +20,12 @@
 /**
- * Interface used to represent both {@link BeaconBlock} and {@link BlockContents} and their blinded
- * variants: <a
+ * Interface used to represent both {@link BeaconBlock} and {@link BlockContentsDeneb} and their
```
and `{@link BlockContentsFulu}`

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/BlockContainerSchema.java:
```diff
@@ -14,11 +14,11 @@
- * Interface used to represent both {@link BeaconBlockSchema} and {@link BlockContentsSchema} and
- * their blinded variants
+ * Interface used to represent both {@link BeaconBlockSchema} and {@link BlockContentsSchemaDeneb}
+ * and their blinded variants
```
and BlockContentsSchemaFulu

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/SignedBlockContainer.java:
```diff
@@ -20,12 +20,12 @@
- * Interface used to represent both {@link SignedBeaconBlock} and {@link SignedBlockContents} and
- * their blinded variants: <a
+ * Interface used to represent both {@link SignedBeaconBlock} and {@link SignedBlockContentsDeneb}
+ * and their blinded variants: <a
```
and `SignedBlockContentsFulu`

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/SignedBlockContainerSchema.java:
```diff
@@ -14,11 +14,11 @@
 /**
  * Interface used to represent both {@link SignedBeaconBlockSchema} and {@link
- * SignedBlockContentsSchema} and their blinded variants
+ * SignedBlockContentsSchemaDeneb} and their blinded variants
```
and SignedBlockContentsSchemaFulu

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/SlotAndBlockRoot.java:
```diff
@@ -30,6 +31,14 @@ public SlotAndBlockRoot(final UInt64 slot, final Bytes32 blockRoot) {
     this.blockRoot = blockRoot;
   }
 
+  public static SlotAndBlockRoot createLow(final UInt64 slot) {
```
Is it used anywhere?

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/SlotAndBlockRoot.java:
```diff
@@ -30,6 +31,14 @@ public SlotAndBlockRoot(final UInt64 slot, final Bytes32 blockRoot) {
+    return new SlotAndBlockRoot(slot, Bytes32.ZERO);
+  }
+
+  public static SlotAndBlockRoot createHigh(final UInt64 slot) {
```
Is it used anywhere?

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/util/BlobsUtil.java:
```diff
@@ -75,16 +77,25 @@ public Bytes generateRawBlobTransactionFromKzgCommitments(
     return kzg.computeBlobKzgProof(blob.getBytes(), kzgCommitment);
   }
 
+  public List<KZGProof> computeKzgCellProofs(final Blob blob) {
```
maybe mark it as `@VisibleForTests` for clarification

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/executionlayer/ExecutionLayerChannel.java:
```diff
@@ -124,7 +132,10 @@ SafeFuture<ForkChoiceUpdatedResult> engineForkChoiceUpdated(
+  SafeFuture<List<Optional<BlobAndProof>>> engineGetBlobAndProofs(
+      List<VersionedHash> blobVersionedHashes, UInt64 slot);
+
+  SafeFuture<List<BlobAndCellProofs>> engineGetBlobAndCellProofsList(
```
I don't like this interface difference. From the spec we either get all or nothing here. Maybe it should be wrapped with `Optional<>` too? I'd at least create an issue to revisit

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/results/FailedBlockImportResult.java:
```diff
@@ -55,6 +55,11 @@ public boolean hasFailedExecutingExecutionPayload() {
   }
 
+  @Override
+  public boolean isDataNotYetAvailable() {
```
do we need `Yet` here and previously in `BatchImportResult`?

zilm13 commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blocks/BlockContentsWithBlobsSchema.java:
```diff
@@ -0,0 +1,37 @@
+import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.Blob;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGProof;
+
+public interface BlockContentsWithBlobsSchema<T extends BlockContainer>
```
BlockContents are always with Blobs, may worth renaming this and Signed container

zilm13 commented on ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/fulu/helpers/MiscHelpersFuluTest.java:
```diff
@@ -0,0 +1,300 @@
+  @MethodSource("getExtendedSampleCountFixtures")
+  public void getExtendedSampleCountReturnsCorrectValues(
+      final int allowedFailures, final int numberOfSamples) {
+    assertThat(miscHelpersFulu.getExtendedSampleCount(UInt64.valueOf(allowedFailures)))
```
I'd clarify in `MiscHelpersFulu` that called method that it's not actually used anymore to avoid any confusion. It was designed for lossy sampling which is not a part of the spec anymore. But called helper is still part of the spec surprisingly

zilm13 commented on ethereum/spec/src/testFixtures/java/tech/pegasys/teku/spec/TestSpecFactory.java:
```diff
@@ -492,19 +499,17 @@ private static SpecConfigAndParent<? extends SpecConfig> getFuluSpecConfig(
+            Eth2Network.MINIMAL, capellaForkEpoch, denebForkEpoch, electraForkEpoch, fuluForkEpoch);
+    return create(config, SpecMilestone.FULU);
   }
 
```
I'd restore removed javadoc here and change ELECTRA to FULU:
```java
  // Our current config files contain FULU params.
  // So all specConfigs created from them will be FULU.
  // Here we just want to make sure that a given config supports the given milestone
  // (which useless in theory because they are all FULU)

```

zilm13 reviewed (commented):
First half of the review

lucassaldanha committed:
PR comments

zilm13 committed:
Add logging on DataColumnSidecars recovery failure (#9486)

lucassaldanha committed:
Merge branch 'master' into master-das

lucassaldanha commented on beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/AbstractBlockPublisher.java:
```diff
@@ -163,7 +236,14 @@
+  abstract void publishDataColumnSidecars(
+      List<DataColumnSidecar> dataColumnSidecars,
+      SignedBeaconBlock block,
+      BlockPublishingPerformance blockPublishingPerformance);
```
This is ok. It isn't used because we have an associated TODO for it. We should keep the signature as is.

lucassaldanha commented on ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/util/BlobsUtil.java:
```diff
@@ -75,16 +77,25 @@
   }
 
+  public List<KZGProof> computeKzgCellProofs(final Blob blob) {
+    return kzg.computeCellsAndProofs(blob.getBytes()).stream().map(KZGCellAndProof::proof).toList();
```
This is used for tests.

lucassaldanha commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetrieverTest.java:
```diff
@@ -0,0 +1,223 @@
+        Stream.generate(dataStructureUtil::randomValidBlob).limit(blobCount).toList();
+    BeaconBlock block = blockResolver.addBlock(10, blobCount);
+    List<DataColumnSidecar> sidecars =
+        miscHelpers.constructDataColumnSidecarsOld(createSigned(block), blobs, kzg);
```
used on tests

lucassaldanha commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetrieverTest.java:
```diff
@@ -0,0 +1,223 @@
+        Stream.generate(dataStructureUtil::randomValidBlob).limit(blobCount).toList();
+    BeaconBlock block_10_0 = blockResolver.addBlock(10, blobCount);
+    List<DataColumnSidecar> sidecars_10_0 =
+        miscHelpers.constructDataColumnSidecarsOld(createSigned(block_10_0), blobs_10_0, kzg);
```
used on tests

lucassaldanha commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetrieverTest.java:
```diff
@@ -0,0 +1,223 @@
+        Stream.generate(dataStructureUtil::randomValidBlob).limit(blobCount).toList();
+    BeaconBlock block = blockResolver.addBlock(10, blobCount);
+    List<DataColumnSidecar> sidecars =
+        miscHelpers.constructDataColumnSidecarsOld(createSigned(block), blobs, kzg);
```
used on tests

lucassaldanha commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/RecoveringSidecarRetrieverTest.java:
```diff
@@ -0,0 +1,223 @@
+        Stream.generate(dataStructureUtil::randomValidBlob).limit(blobCount).toList();
+    BeaconBlock block_10_1 = blockResolver.addBlock(10, blobCount);
+    List<DataColumnSidecar> sidecars_10_1 =
+        miscHelpers.constructDataColumnSidecarsOld(createSigned(block_10_1), blobs_10_1, kzg);
```
used on tests

lucassaldanha commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/SimpleSidecarRetrieverTest.java:
```diff
@@ -0,0 +1,306 @@
+    final List<Blob> blobs = Stream.generate(dataStructureUtil::randomValidBlob).limit(1).toList();
+    final BeaconBlock block = blockResolver.addBlock(10, 1);
+    final List<DataColumnSidecar> sidecars =
+        miscHelpers.constructDataColumnSidecarsOld(createSigned(block), blobs, kzg);
```
used on tests

lucassaldanha commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/retriever/SimpleSidecarRetrieverTest.java:
```diff
@@ -0,0 +1,306 @@
+
+  @Test
+  void performanceTest() {
+    final List<TestPeer> testNodes =
+        craftNodeIds()
+            .map(
+                nodeId ->
+                    new TestPeer(stubAsyncRunner, nodeId, Duration.ofMillis(100))
+                        .currentRequestLimit(1000))
+            .limit(128)
+            .peek(node -> custodyCountSupplier.setCustomCount(node.getNodeId(), columnCount))
+            .peek(testPeerManager::connectPeer)
+            .toList();
```
@zilm13 this unit test `performanceTest()` seems a bit odd. Help me understand what we are doing here?

lucassaldanha committed:
CodeQL suggestions

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/retriever/DataColumnSidecarRetriever.java:
```diff
@@ -0,0 +1,62 @@
+   * The request may complete with this exception when requested column is no more on our local
+   * canonical chain
+   */
+  class NotOnCanonicalChainException extends RuntimeException {
```
maybe `IllegalStateException` is more appropriate for it?

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java:
```diff
@@ -0,0 +1,389 @@
+              .toList();
+    }
+
+    LOG.info(
```
not `info` for production,  it happens on every slot

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java:
```diff
@@ -0,0 +1,389 @@
+        .engineGetBlobAndCellProofsList(versionedHashes, slotAndBlockRoot.getSlot())
+        .thenAccept(
+            blobAndCellProofsList -> {
+              LOG.info("Found {} blobs", blobAndCellProofsList.size());
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java:
```diff
@@ -0,0 +1,389 @@
+                  versionedHashes.size(),
+                  blobAndCellProofsList.size());
+
+              LOG.info(
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java:
```diff
@@ -0,0 +1,389 @@
+  }
+
+  private void logLocalElBlobsLookupFailure(final Throwable error) {
+    LOG.warn("Local EL blobs lookup failed: {}", getRootCauseMessage(error));
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/CustodyGroupCountManagerImpl.java:
```diff
@@ -0,0 +1,168 @@
+    if (oldCustodyGroupSyncedCount == custodyGroupSyncedCount) {
+      return;
+    }
+    LOG.info("Synced custody group count updated to {}.", custodyGroupSyncedCount);
```
It's a rare event but still not sure if we should keep it in info for production

lucassaldanha replied:
I'd say we keep it as debug and if we feel like it is useful during tests we bump it up later.

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/CustodyGroupCountManagerImpl.java:
```diff
@@ -0,0 +1,168 @@
+  private synchronized void updateCustodyGroupCount(final int newCustodyGroupCount) {
+    final int oldCustodyGroupCount = custodyGroupCount.getAndSet(newCustodyGroupCount);
+    if (oldCustodyGroupCount != newCustodyGroupCount) {
+      LOG.info(
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasCustodySync.java:
```diff
@@ -0,0 +1,201 @@
+
+  public DasCustodySync(
+      final DataColumnSidecarCustody custody, final DataColumnSidecarRetriever retriever) {
+    this(custody, retriever, 10 * 1024, 2 * 1024);
```
magic numbers may need explanation, but I'm not sure why we have these numbers actually

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasLongPollCustody.java:
```diff
@@ -0,0 +1,188 @@
+  @Override
+  public SafeFuture<Boolean> hasCustodyDataColumnSidecar(
+      final DataColumnSlotAndIdentifier columnId) {
+    SafeFuture<Optional<Boolean>> pendingFuture =
```
finals please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasLongPollCustody.java:
```diff
@@ -0,0 +1,188 @@
+      final DataColumnSlotAndIdentifier columnId) {
+    SafeFuture<Optional<Boolean>> pendingFuture =
+        addPendingRequest(columnId).thenApply(maybeSidecar -> maybeSidecar.map(__ -> true));
+    SafeFuture<Optional<Boolean>> existingFuture =
```
finals please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasPreSampler.java:
```diff
@@ -0,0 +1,70 @@
+    final List<SignedBeaconBlock> blocksToSample =
+        blocks.stream().filter(this::isSamplingRequired).toList();
+
+    LOG.info(
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasSamplerBasic.java:
```diff
@@ -0,0 +1,204 @@
+                    .sorted()
+                    .toList();
+
+            LOG.info(
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DasSamplerBasic.java:
```diff
@@ -0,0 +1,204 @@
+                  .thenPeek(
+                      retrievedColumns -> {
+                        if (!retrievedColumns.isEmpty()) {
+                          LOG.info(
```
downgrade level please

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarCustodyImpl.java:
```diff
@@ -0,0 +1,268 @@
+      // implement in future invalidating current custody as number of required groups have
+      // increased (https://github.com/Consensys/teku/issues/9468)
+      if (groupCount > oldGroupCount) {
+        LOG.info("Custody group count changed from {} to {}", oldGroupCount, groupCount);
```
Info logging here and the other below duplicates custody group count manager logging. We don't need both in info. We could move this to debug or one in manager to debug

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarCustodyImpl.java:
```diff
@@ -0,0 +1,268 @@
+                          //  non-finalized epochs could be still not synced with up-to-date custody
+                          // (https://github.com/Consensys/teku/issues/9469)
+                          if (firstIncompleteOrLastComplete.slot().equals(firstNonFinalizedSlot)) {
+                            LOG.info(
```
this one is the other

zilm13 commented on ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java:
```diff
@@ -466,11 +478,15 @@ private SafeFuture<BlockImportResult> onBlock(
                 result -> {
+                  LOG.trace(
+                      "Availability check slot {} result: {}",
+                      block.getSlot(),
```
I'd log slotBlockRoot

zilm13 commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/blobs/BlobSidecarManagerTest.java:
```diff
@@ -206,7 +205,7 @@ void createAvailabilityChecker_shouldReturnANotRequiredAvailabilityCheckerWhenBl
 
     assertThat(blobSidecarManager.createAvailabilityChecker(block))
-        .isEqualTo(BlobSidecarsAvailabilityChecker.NOT_REQUIRED);
+        .isEqualTo(AvailabilityChecker.NOOP_BLOBSIDECAR);
```
maybe update test name in this case

zilm13 commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/DasLongPollCustodyTest.java:
```diff
@@ -0,0 +1,253 @@
+  }
+
+  @Test
+  void testEmptyIsNotRetrunedImmediatelyAtBeginningOfCurrentSlot() {
```
NotReturned

zilm13 commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/validation/DataColumnSidecarGossipValidatorTest.java:
```diff
@@ -0,0 +1,472 @@
+  }
+
+  @TestTemplate
+  void shouldIgnoreWhenParentIsNotAvailable_slot() {
```
let's rename in java-case

zilm13 commented on infrastructure/logging/src/main/resources/log4j2.xml:
```diff
@@ -1,6 +1,13 @@
 <Configuration level="INFO">
     <Loggers>
+        <!--         Remove all DEBUG and TRACE logging from noisy dependencies-->
+        <Logger name="io.netty" level="INFO" additivity="FALSE"/>
```
It affects default logging behavior (eliminating 3rd libraries debug+ logging from our logs). @rolfyone do you confirm this change?

lucassaldanha replied:
Let's revert it and do it later if we feel like we want to.

zilm13 commented on teku/src/main/java/tech/pegasys/teku/cli/options/InteropOptions.java:
```diff
@@ -61,6 +62,14 @@ public class InteropOptions {
       description = "Represents the total number of validators in the network")
   private int interopNumberOfValidators = InteropConfig.DEFAULT_INTEROP_NUMBER_OF_VALIDATORS;
 
+  @Option(
```
Implementation of this option was dropped. We could have a separate issue for removal.

lucassaldanha replied:
https://github.com/Consensys/teku/issues/9494

zilm13 commented on validator/client/src/main/java/tech/pegasys/teku/validator/client/AttestationDutySchedulingStrategy.java:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright Consensys Software Inc., 2025
+ * Copyright Consensys Software Inc., 2024
```
Intentional change?

zilm13 commented on storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbInstanceFactory.java:
```diff
@@ -147,6 +147,8 @@ private static DBOptions createDBOptions(
             .setMaxBackgroundJobs(configuration.getMaxBackgroundJobs())
             .setDbWriteBufferSize(configuration.getWriteBufferCapacity())
             .setMaxOpenFiles(configuration.getMaxOpenFiles())
+            .setBytesPerSync(1_048_576L) // 1MB
```
both these settings should be removed as they were intentionally removed from the master and restored here by merge error

lucassaldanha replied:
Nice catch!

zilm13 commented on services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java:
```diff
@@ -1193,6 +1576,11 @@ protected void initP2PNetwork() {
+    this.nodeId =
+        p2pNetwork
+            .getDiscoveryNodeId()
+            .orElseThrow(() -> new InvalidConfigurationException("NodeID is required"));
```
Maybe something more friendly to user, say "Failed to get NodeId from Discovery System"

zilm13 commented on ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/BlobSidecarsAvailabilityCheckerTest.java:
```diff
@@ -0,0 +1,345 @@
+import tech.pegasys.teku.storage.client.RecentChainData;
+import tech.pegasys.teku.storage.store.UpdatableStore;
+
+public class BlobSidecarsAvailabilityCheckerTest {
```
it duplicates `ForkChoiceBlobSidecarsAvailabilityCheckerTest`. Remove this one, leave `ForkChoiceBlobSidecarsAvailabilityCheckerTest`

zilm13 reviewed (commented):
Second and final part of the review
1. I'm not sure on info, warn, error logging downgrade suggestions. If doesn't come to EventLogger, may be we could keep it? I've started commeinting on it but not sure now. There were several other info logs left that I had not commented (for example, in `DataColumnSidecarAvailabilityChecker`, `GossipFailureLogger` etc). They should be addressed if we decide to downgrade them all out of info. 
2. We may need to add finals everywhere, especially finals are missing in tests and testFixtures

lucassaldanha committed:
2nd round of PR comments

lucassaldanha committed:
Merge branch 'master' into master-das

lucassaldanha committed:
Fix constructor

lucassaldanha committed:
Other log downgrades

lucassaldanha committed:
Decrease log
